magenta-lib/src/main/scala/magenta/deployment_type/CloudFormationDeploymentTypeParameters.scala

```diff
@@ -156,12 +156,25 @@ trait CloudFormationDeploymentTypeParameters {
       target: DeployTarget,
       reporter: DeployReporter,
       lookup: Lookup
-  ): String => String => Map[String, String] => Option[String] = {
-    accountNumber =>
-      if (amiEncrypted(pkg, target, reporter)) {
-        lookup.getLatestAmi(Some(accountNumber), _ => true)
-      } else {
-        lookup.getLatestAmi(None, unencryptedTagFilter)
+  ): CfnParam => String => String => Map[String, String] => Option[String] = {
+    amiCfnParam =>
+      { accountNumber =>
+        {
+          val useEncryptedAmi: Boolean = amiParametersToTags.get(pkg) match {
+            case Some(params) =>
+              params
+                .getOrElse(amiCfnParam, Map.empty)
+                .get("Encrypted")
+                .contains("true")
```

Comment:
do we need a toLowercase in here first for a safe comparison to the string `"true"`
```suggestion
                .map(_.toLowercase)
                .contains("true")
```

Reply:
I'm tempted to not do this, as [we don't elsewhere](https://github.com/guardian/riff-raff/blob/main/magenta-lib/src/main/scala/magenta/deployment_type/CloudFormationDeploymentTypeParameters.scala#L14-L22), and this tag shouldn't be manually set, so should be reliably cased. WDYT?

Reply:
yeah fair enough, although what about using https://www.scala-lang.org/files/archive/api/2.13.x/scala/collection/StringOps.html#toBooleanOption:Option[Boolean] then pattern match or something?

Reply:
Oooh! IIUC that'll become:

```scala
params
  .getOrElse(amiCfnParam, Map.empty)
  .getOrElse("Encrypted", "false") // default to `"false"` if no `Encrypted` tag
  .toBooleanOption.getOrElse(false) // default to `false` if failed to parse the tag's string value
```

This form doesn't read as easy to me, though I might be missing a more terse implementation.

Reply:
or 
```
params
  .getOrElse(amiCfnParam, Map.empty)
  .get("Encrypted")
  .flatMap(_.toBooleanOption)
  .getOrElse(false)
```

agreed its certainly no more readable than what you had

Reply:
`toBooleanOption` made fixing a regression really nice, thanks! https://github.com/guardian/riff-raff/pull/1013